src/function_manager.jl

```diff
@@ -55,19 +55,26 @@ end
 # so "jlfun" seems reasonable
 const FUNCTION_MANAGER_NAMESPACE = "jlfun"
 
-Base.@kwdef struct FunctionManager
+Base.@kwdef mutable struct FunctionManager
     gcs_client::ray_jll.JuliaGcsClient
     functions::Dict{String,Any}
+
+    function FunctionManager(gcs_client, functions)
+        fm = new(gcs_client, functions)
+        f(x) = ray_jll.Disconnect(x.gcs_client)
+        return finalizer(f, fm)
+    end
 end
```

Comment:
I don't think we need the `finalizer` at all here now.
```suggestion
Base.@kwdef struct FunctionManager
    gcs_client::ray_jll.JuliaGcsClient
    functions::Dict{String,Any}
end
```

Reply:
ah yeah forgot to remove it

Reply:
pretty sure the `FunctionManager` needs to be mutable